File: backend-spring/src/main/java/com/sourcelens/module/scantask/entity/ScanTask.java
package com.sourcelens.module.scantask.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("scan_tasks")
public class ScanTask {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private Long repositoryId;

    private String branch;

    private String commitSha;

    private String status;

    private String activeLockKey;

    private String triggerType;

    private LocalDateTime startedAt;

    private LocalDateTime finishedAt;

    private String errorMessage;

    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Boolean deleted;
}

File: web-console/src/api/scanTask.ts
import client from './client'
import type { Result } from './client'

export interface ScanTask {
  id: number
  projectId: number
  repositoryId: number
  branch: string
  commitSha: string | null
  status: string
  triggerType: string
  startedAt: string | null
  finishedAt: string | null
  errorMessage: string | null
  createdBy: number
  createdAt: string
}

export interface PageResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
}

export const scanTaskApi = {
  create: (repositoryId: number, data: { projectId: number; branch?: string }) =>
    client.post<Result<ScanTask>>(`/repositories/${repositoryId}/scan-tasks`, data),
  list: (projectId: number, page = 1, pageSize = 20) =>
    client.get<Result<PageResult<ScanTask>>>(`/projects/${projectId}/scan-tasks`, { params: { page, pageSize } }),
  detail: (id: number) =>
    client.get<Result<ScanTask>>(`/scan-tasks/${id}`),
  cancel: (id: number) =>
    client.post<Result<ScanTask>>(`/scan-tasks/${id}/cancel`),
}
